### ISSUE TITLE
Table: allow to convert a Row into a dict and/or use it in str.format()

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#10967]
### Description

While I can convert a complete `Table` into a dict (column name as key, column data array as value), this doesn't work with a table row. I also can't use the table row to format a string:
```python
from astropy.table import Table

tbl = Table({'num': [2, 4, 6], 'what': ['two', 'four', 'six']})

table_dict = dict(tbl)   # this works

row_dict = dict(tbl[0])  # but this doesn't

for row in tbl:
    print('What: {what} is {num:2i}'.format(**row))  # This would be useful
```
Or is there another simple solution to format table rows?

**Comment by pllim** [comment]
Works on 4.3.dev for me.

```python
>>> dict(tbl[0])                                                                                
{'num': 2, 'what': 'two'}
>>> for row in tbl: 
...     print('What: {what} is {num:2d}'.format(**row))                                                                                             
What: two is  2
What: four is  4
What: six is  6
```

What version of `astropy` are you using? How is it not working for you?

**Comment by olebole** [comment]
I had 4.0.1, but I see it fixed with 4.1. 
So, the time to implement the feature request was actually negative :-) 
Sorry, and thank you!
